python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -240,6 +285,39 @@ def evaluate(
             df_group_metrics = df_grouped_stats.apply(
                 metrics_helper.summarize_grouped_stats, axis=1
             )
+            # Add power analysis to group metric dataframe
+            if len(group_keys) > 0 and len(eval_dict[RankingConstants.METRICS]) > 0:
+                df_group_metrics = df_group_metrics.reset_index()
+                if len(group_keys) > 1:
+                    df_group_metrics[str(group_keys)] = df_group_metrics[group_keys].apply(tuple, axis=1)
+                df_group_metrics[str(group_keys)] = df_group_metrics[group_keys]
+                df_group_metrics = pd.merge(df_group_metrics, group_metrics_stat_sig, on=str(group_keys), how='left').drop(
+                    columns=[str(group_keys)])
+
+            # Writing stat sig. groupwise results per metric
+            for metric in eval_dict[RankingConstants.METRICS]:
+                stat_sig_df, improved, degraded, stat_sig_groupwise_metric_improv = metrics_helper.generate_stat_sig_based_metrics(
+                    df_group_metrics, metric, group_keys)
+                improv_count = len(improved)
+                degrad_count = len(degraded)
+                if logs_dir:
+                    self.file_io.write_df(
+                        stat_sig_df,
+                        outfile=os.path.join(
+                            logs_dir, "stat_sig_" + metric + '_' + RelevanceModelConstants.GROUP_METRICS_CSV_FILE),
+                    )
+                # Compute group metrics summary
+                stat_sig_df_summary = stat_sig_df.describe(include='all')
+                self.logger.info(
+                    "\nComputing group metrics for {} stat. sig. lifts using keys: {}".format(metric,
+                                                                                              group_keys))
+                self.logger.info("\nGroupwise Metrics for {}: \n{}".format(metric, stat_sig_df_summary.T))
+                metrics_dict["stat_sig_" + metric + "_improved_groups"] = improv_count
+                metrics_dict["stat_sig_" + metric + "_degraded_groups"] = degrad_count
+                metrics_dict["stat_sig_" + metric + "_group_improv_perc"] = stat_sig_groupwise_metric_improv
+                metrics_dict["stat_sig_improved_" + metric + "_groups"] = sorted(improved)
+                metrics_dict["stat_sig_degraded_" + metric + "_groups"] = sorted(degraded)
```

Comment:
@mohazahran This should definitely not be in `ranking_model`. All this "summarization" should be in the helpers.